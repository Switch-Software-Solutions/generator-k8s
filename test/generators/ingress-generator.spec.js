const IngressGenerator = require('../../generators/ingress');
const { readFile, pathExists } = require('fs-extra');
const { join } = require('path');
const helpers = require('yeoman-test');

const defaultPrompts = {
  'global-name': 'resource',
  'ingress-name': '${CI_PROJECT_PATH_SLUG}',
  'ingress-tls-name': '${CI_PROJECT_PATH_SLUG}',
  'ingress-url': 'smartive.ch;www.smartive.ch',
  'ingress-service-port': 80,
};

describe('Resource Generator: Ingress', () => {

  it('should generate a resource file', async () => {
    const folder = await helpers
      .run(IngressGenerator)
      .withPrompts(defaultPrompts);
    expect(await pathExists(join(folder, 'resource.yml'))).toBe(true);
  });

  it('should generate a host for a single url', async () => {
    const folder = await helpers
      .run(IngressGenerator)
      .withPrompts({
        ...defaultPrompts,
        'ingress-url': 'smartive.ch',
      });
    const content = await readFile(join(folder, 'resource.yml'), 'utf8');
    expect(content).toMatchSnapshot();
  });

//  it('should generate a host for multiple urls', async () => {
//    const folder = await helpers
//      .run(IngressGenerator)
//      .withPrompts({
//        ...defaultPrompts,
//        'ingress-url': 'smartive.ch;www.smartive.ch',
//      });
//    const content = await readFile(join(folder, 'resource.yml'), 'utf8');
//    expect(content).toMatchSnapshot();
//  });

//  it('should not generate a host for an empty url', async () => {
//    const folder = await helpers
//      .run(IngressGenerator)
//      .withPrompts({
//        ...defaultPrompts,
//        'ingress-url': 'smartive.ch;www.smartive.ch;;',
//      });
//    const content = await readFile(join(folder, 'resource.yml'), 'utf8');
//    expect(content).toMatchSnapshot();
//  });

});
