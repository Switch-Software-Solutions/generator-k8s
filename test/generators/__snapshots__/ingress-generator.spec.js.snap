// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource Generator: Ingress should generate a host for a single url 1`] = `
"# Ingress
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: \${CI_PROJECT_PATH_SLUG}-ingress
  annotations:
    kubernetes.io/ingress.class: nginx
    nginx.ingress.kubernetes.io/rewrite-target: /
    nginx.ingress.kubernetes.io/ssl-redirect: 'true'

spec:
  tls:
    - secretName: \${CI_PROJECT_PATH_SLUG}
      hosts:
        - smartive.ch
  rules:
    - host: smartive.ch
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: \${CI_PROJECT_PATH_SLUG}-cluster-ip-service
                port:
                  number: 80
"
`;
